### PR TITLE
Refact asset-manager for simplicity

### PR DESCRIPTION
**Pull request by yrong**

### CODE REVIEW

**Comment by GopherJ** [comment]
@yrong I created checklist at: #1578

File: pallets/asset-registry/src/lib.rs [hunk 1]
```diff
@@ -188,31 +166,14 @@ pub mod pallet {
...
         ) -> DispatchResult {
             T::AssetModifierOrigin::ensure_origin(origin)?;
 
-            let asset_exist = asset_id.is_some();
-            let asset_id: T::AssetId = match asset_id {
-                Some(asset_id) => asset_id,
-                None => asset.clone().into(),
-            };
             ensure!(
                 AssetIdType::<T>::get(&asset_id).is_none(),
```
Comment (GopherJ) [inline pallets/asset-registry/src/lib.rs:174]: contains_key maybe simpler

File: pallets/asset-registry/src/lib.rs [hunk 2]
```diff
@@ -188,31 +166,14 @@ pub mod pallet {
         pub fn register_asset(
             origin: OriginFor<T>,
             asset: T::AssetType,
-            asset_id: Option<T::AssetId>,
-            metadata: Option<T::AssetRegistrarMetadata>,
-            min_amount: T::Balance,
-            is_sufficient: bool,
+            asset_id: T::AssetId,
         ) -> DispatchResult {
             T::AssetModifierOrigin::ensure_origin(origin)?;
```
Comment (GopherJ) [inline pallets/asset-registry/src/lib.rs:171]: AssetModifierOrigin can be simplified to `UpdateOrigin`

**Review by GopherJ — APPROVED** [review]

File: pallets/asset-registry/src/lib.rs (hunk 2 above)
Comment (yrong) [reply pallets/asset-registry/src/lib.rs]: [f858a76](https://github.com/parallel-finance/parallel/pull/1581/commits/f858a76188c11ec088e351ed3876845c45b66614)

**Pull request merged**